pennylane/interfaces/batch/autograd.py

```diff
@@ -158,7 +158,7 @@ def grad_fn(dy):
         """Returns the vector-Jacobian product with given
         parameter values and output gradient dy"""
 
-        dy = dy[0]
+        dy = [qml.math.T(d) for d in dy[0]]

```

Comment:
Fixes the bug where the Autograd interface was returning the Jacobian transposed.

Note that transposing the initial `dy` vectors is safer than transposing the computed Jacobian. This is because:

- the `dy` vector will always be one-dimensional arrays (since QNodes have restricted output), so the transpose is well defined.

- The computed Jacobian, however, may be N-dimensional, since the shape is dependent on both the QNode output _and_ the QNode input weights, which can be arbitrary. So transposing the computed Jacobian is not well defined --- different autodiff frameworks may differ in how they swap the axes of a N-dimensional tensor.

Reply:
The difference in ordering convention is... concerning :worried:

Reply:
How were the old tests passing if the transpose was being returned?

Reply:
The tests were assuming the transposed version was correct 😆 I wasn't sure which transposition was the correct one at the time